psycop/common/model_training/training/train_and_predict.py

```diff
@@ -88,6 +93,74 @@ def stratified_cross_validation(
     return train_df
 
 
+def multilabel_cross_validation(
+    cfg: FullConfigSchema,
+    pipe: Pipeline,
+    train_df: pd.DataFrame,
+    train_col_names: list[str],
+    outcome_col_name: list[str],
+) -> pd.DataFrame:
+    """Performs stratified and grouped cross validation using the pipeline."""
+    if cfg.model.name not in SUPPORTS_MULTILABEL_CLASSIFICATION:
+        raise ValueError(
+            f"{cfg.model.name} does not support multilabel classification. Models that support multilabel classification include: {SUPPORTS_MULTILABEL_CLASSIFICATION}.",
+        )
+
+    msg = Printer(timestamp=True)
+
+    X = train_df[train_col_names]
+    y = train_df[outcome_col_name]
+
+    # Create folds
+    msg.info("Creating folds")
+    msg.info(f"Training on {X.shape[1]} columns and {X.shape[0]} rows")
+
+    # stratify by first outcome column
+    folds = StratifiedGroupKFold(n_splits=cfg.n_crossval_splits).split(
+        X=X,
+        y=y[outcome_col_name[0]],
+        groups=train_df[cfg.data.col_name.id],
+    )
```

Comment:
to stratify across all outcomes: concanate the output columns into a single column (an example row would be e.g. "0001" - stratify on this column